### PR TITLE
Handle grapheme clusters

### PR DESCRIPTION
**Pull request by rdxdkr**
Given that your library deals primarily with strings that are words from possibly different languages, I thought it would only be natural to add support for complex grapheme clusters. This way you can abbreviate words from virtually any language supported by UTF-8 and the library can be used by anyone in the world.

The idea came to me after seeing that you used to count the chars by calling `.chars().count()`, which certainly works for strings written in plain Latin alphabet like what I'm using right now. However, when you try it on certain foreign languages it produces unexpected results and it doesn't work as you might have intended, as explained [here](https://doc.rust-lang.org/book/ch08-02-strings.html#indexing-into-strings). Make sure to check out the [library](https://docs.rs/unicode-segmentation/latest/unicode_segmentation/) I've used, which is opt-in instead of being part of the language itself.

### CODE REVIEW

**Comment by Vagelis-Prokopiou** [comment]
Hi @rdxdkr.

This is a nice thought indeed. It is useful for sure.

I have a couple of thoughts though.

1) Is the unicode-segmentation the goto library for this job? I did not check it through myself (yet). At least it seems that it is well tested . I downloaded it and run the tests.

2) ~~This library (unicode-segmentation) has a lot of dependencies in itself. Therefore, I would like this functionality to be available through a feature flag. I would not like all of this extra code to be build and included in the executable by default. E.g. "non_ascii" feature flag~~.

Basically, I missed that all the dependencies are dev dependencies. So point 2 is invalid. Therefore it seems that not much is needed in order to merge. Check only my comment on the test.

**Comment by rdxdkr** [comment]
Yes, this specific library seems to be the preferred one, and probably even the only one that is still actively maintained (it got its latest update just yesterday).
